addons/web_editor/static/src/components/html_history_dialog/html_history_dialog.js

```diff
@@ -0,0 +1,124 @@
+const { DateTime } = luxon;
+
+class HtmlHistoryDialog extends Component {
+
+    state = useState({
+        latestHistoryDiffs: [],
+        currentHtml: null,
+        currentComparisonHtml: null,
+        currentId: null,
+        buttonDisabled: true,
+    });
+    setup() {
+        this.size = 'xl';
+        this.title = _t("Recent History");
+        this.orm = useService("orm");
+
+        this.comparisonCache = {};
+        this.htmlCache = {};
+
+        onMounted(() => this.init());
+    }
+
+    async init(limit=100) {
+        // limit the number of record and order them by id desc
+        const recIds = this.props.recordsIds.slice(0, limit).reverse();
+
+        // get record data
+        this.state.latestHistoryDiffs = await this.orm.call(
+            'field.html.history.diff',
+            'read',
+            [recIds, ["id", "create_date", "create_uid"]]
+        );
+
+        // get the comparison html for the latest diff
+        this.state.buttonDisabled = false;
+        await this.updateCurrentComparisonHtml(this.state.latestHistoryDiffs[0].id);
+    }
+
+    async updateCurrentComparisonHtml(historyDiffId) {
+        if (this.state.currentId === historyDiffId || this.state.buttonDisabled) {
+            return;
+        }
+        this.state.buttonDisabled = true;
+        this.state.currentId = historyDiffId;
+        this.state.currentComparisonHtml = await this.getComparisonAtDiffId(historyDiffId);
+        this.state.currentHtml = await this.getHtmlAtDiffId(historyDiffId);
+        this.state.buttonDisabled = false;
+    }
+
+    async getComparisonAtDiffId(historyDiffId) {
+        if (!this.comparisonCache[historyDiffId]) {
+            const comparison = await this.orm.call(
+                "field.html.history.diff",
+                "get_comparison",
+                [historyDiffId]
+            );
+            this.comparisonCache[historyDiffId] = markup(comparison);
+        }
+        return this.comparisonCache[historyDiffId];
+    }
+    async getHtmlAtDiffId(historyDiffId) {
+        if (!this.htmlCache[historyDiffId]) {
+            const restoredVersion = await this.orm.call(
+                "field.html.history.diff",
+                "get_version",
+                [historyDiffId]
+            );
+            this.htmlCache[historyDiffId] = markup(restoredVersion);
+        }
+        return this.htmlCache[historyDiffId];
+    }
+
+    async _onHistoryRestoreClick() {
+        this.state.buttonDisabled = true;
+        await this.restoreVersion(this.state.currentId);
+        this.props.close();
+    }
+    async restoreVersion(historyDiffId) {
+        const restoredVersion = this.getHtmlAtDiffId(historyDiffId);
+        this.props.restoreRequested(restoredVersion);
+    }
+
+    /**
+     * Getters
+     **/
+
+    getDiffAuthor(rec) {
+        if (rec.create_uid && rec.create_uid[1]) {
+            return rec.create_uid[1];
+        }
+        return 'Unknown';
+    }
```

Comment:
This is used only once and is a one-liner: `rec.create_uid?.[1] ?? 'Unknown'`. I'm not even sure we need to handle the case where there is no create_uid as I think it's a mandatory field, or at least it's filled automatically.